src/ompl/geometric/planners/quotientspace/datastructures/src/QuotientSpace.cpp

```diff
@@ -0,0 +1,1449 @@
+            unsigned int N = Q1_RN->getDimension();
+
+            ob::StateSpacePtr SO2(new ob::SO2StateSpace());
+            ob::StateSpacePtr RN(new ob::RealVectorStateSpace(N));
+            RN->as<ob::RealVectorStateSpace>()->setBounds(Q1_RN->getBounds());
+
+            X1 = SO2 + RN;
+            X1_dimension_ = 1 + N;
+            break;
+        }
+        case SE3RN_R3:
+        {
+            ob::CompoundStateSpace *Q1_compound = Q1->as<ob::CompoundStateSpace>();
+            const std::vector<StateSpacePtr> Q1_decomposed = Q1_compound->getSubspaces();
+            const std::vector<StateSpacePtr> Q1_SE3_decomposed =
+                Q1_decomposed.at(0)->as<ob::CompoundStateSpace>()->getSubspaces();
+
+            // const ob::SE3StateSpace *Q1_SE3 = Q1_SE3_decomposed.at(0)->as<ob::SE3StateSpace>();
+            // const ob::SO3StateSpace *Q1_SO3 = Q1_SE3_decomposed.at(1)->as<ob::SO3StateSpace>();
+            const ob::RealVectorStateSpace *Q1_RN = Q1_decomposed.at(1)->as<ob::RealVectorStateSpace>();
+            unsigned int N = Q1_RN->getDimension();
+
+            ob::StateSpacePtr SO3(new ob::SO3StateSpace());
+            ob::StateSpacePtr RN(new ob::RealVectorStateSpace(N));
+            RN->as<ob::RealVectorStateSpace>()->setBounds(Q1_RN->getBounds());
+
+            X1 = SO3 + RN;
+            X1_dimension_ = 3 + N;
+            break;
+        }
+        case SE2RN_SE2RM:
+        case SO2RN_SO2RM:
+        case SE3RN_SE3RM:
+        {
+            ob::CompoundStateSpace *Q1_compound = Q1->as<ob::CompoundStateSpace>();
+            const std::vector<StateSpacePtr> Q1_decomposed = Q1_compound->getSubspaces();
+            ob::CompoundStateSpace *Q0_compound = Q0->as<ob::CompoundStateSpace>();
+            const std::vector<StateSpacePtr> Q0_decomposed = Q0_compound->getSubspaces();
+
+            unsigned int N = Q1_decomposed.at(1)->getDimension();
+            unsigned int M = Q0_decomposed.at(1)->getDimension();
+            X1_dimension_ = N - M;
+            X1 = std::make_shared<ob::RealVectorStateSpace>(X1_dimension_);
+
+            RealVectorBounds Q1_bounds =
+                std::static_pointer_cast<ob::RealVectorStateSpace>(Q1_decomposed.at(1))->getBounds();
+            std::vector<double> low;
+            low.resize(X1_dimension_);
+            std::vector<double> high;
+            high.resize(X1_dimension_);
+            RealVectorBounds X1_bounds(X1_dimension_);
+            for (unsigned int k = 0; k < X1_dimension_; k++)
+            {
+                X1_bounds.setLow(k, Q1_bounds.low.at(k + M));
+                X1_bounds.setHigh(k, Q1_bounds.high.at(k + M));
+            }
+            std::static_pointer_cast<ob::RealVectorStateSpace>(X1)->setBounds(X1_bounds);
+            break;
+        }
+        default:
+        {
+            OMPL_ERROR("Unknown QuotientSpace type: %d", type_);
+            exit(0);
```

Comment:
Same as before, throw an error. We don't want to blow up other people's code with internal errors.